backend/commons/src/main/java/org/sonarsource/sonarlint/core/commons/util/git/exceptions/GitException.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * SonarLint Core - Commons
+ * Copyright (C) 2016-2025 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.core.commons.util.git.exceptions;
+
+public class GitException extends IllegalStateException {
```

Comment:
Nitpick: I think it's uncommon to extend `IllegalStateException`, I don't see what benefit it brings. Usually we directly extend `RuntimeException`

Reply:
Because in many places we throw IllegalStateException like in ProcessWrapperFactory or in WinGitUtils, and in corresponding catch blocks we expect the IllegalStateException itself, or something that extends it.